hivemq-edge/src/frontend/src/extensions/datahub/designer/operation/OperationPanel.tsx

```diff
@@ -8,15 +8,28 @@ import useDataHubDraftStore from '@datahub/hooks/useDataHubDraftStore.ts'
 import { ReactFlowSchemaForm } from '@datahub/components/forms/ReactFlowSchemaForm.tsx'
 import { datahubRJSFWidgets } from '@datahub/designer/datahubRJSFWidgets.tsx'
 import { MOCK_OPERATION_SCHEMA } from '@datahub/designer/operation/OperationData.ts'
+import { isFunctionNodeType } from '@datahub/utils/node.utils.ts'
 
 export const OperationPanel: FC<PanelProps> = ({ selectedNode, onFormSubmit }) => {
-  const { nodes, functions } = useDataHubDraftStore()
+  const { nodes, edges, functions } = useDataHubDraftStore()
 
   const formData = useMemo(() => {
     const adapterNode = nodes.find((e) => e.id === selectedNode) as Node<OperationData> | undefined
     if (!adapterNode?.data) return null
+
+    if (adapterNode.data.functionId === OperationData.Function.DATAHUB_TRANSFORM) {
+      const connectedFunctionIds = getIncomers(adapterNode, nodes, edges)
+        .filter(isFunctionNodeType)
+        .map((e) => e.data.name)
+      const gg = adapterNode.data.formData as unknown as OperationData.DataHubTransformType
```

Comment:
suggestion: check on the variable name `gg` might be confusing later

```suggestion
      const gg = adapterNode.data.formData as unknown as OperationData.DataHubTransformType
```